plugin/kubernetes/external.go

```diff
@@ -75,16 +85,52 @@ func (k *Kubernetes) External(state request.Request) ([]msg.Service, int) {
 			continue
 		}
 
-		for _, ip := range svc.ExternalIPs {
-			for _, p := range svc.Ports {
-				if !(matchPortAndProtocol(port, p.Name, protocol, string(p.Protocol))) {
+		if svc.Headless() || endpoint != "" {
+			if endpointsList == nil {
+				endpointsList = endpointsListFunc()
+			}
+			// Endpoint query or headless service
+			for _, ep := range endpointsList {
+				if object.EndpointsKey(svc.Name, svc.Namespace) != ep.Index {
 					continue
 				}
-				rcode = dns.RcodeSuccess
-				s := msg.Service{Host: ip, Port: int(p.Port), TTL: k.ttl}
-				s.Key = strings.Join([]string{zonePath, svc.Namespace, svc.Name}, "/")
 
-				services = append(services, s)
+				for _, eps := range ep.Subsets {
+					for _, addr := range eps.Addresses {
+
+						// See comments in parse.go parseRequest about the endpoint handling.
```

Comment:
What specific comments there are you referring to, and how do they relate to this code?

Reply:
In parse.go is explained how to get each element from a dns name. This specific code was taken from kubernetes.go line n. 543. 

IMHO it's related in the sense of understanding endpointHostname function.

Reply:
OK, how about... 

```suggestion
						// See comments in parse.go parseRequest about endpoint record format.
```

Reply:
I've deleted that comment.